lib/generateClass.js

```diff
@@ -205,6 +213,61 @@ function writeFromJson(def, cw) {
   cw.ln('return inst;');
 }
 
+/**
+ * Generates a JSON serializer for the element
+ * @param {DataElement} def - The definition of the SHR element to generate a serializer for
+ * @param {CodeWriter} cw - The CodeWriter instance to use during generation
+ * @private
+ */
+function writeToJson(def, cw) {
+  // If the element is an Entry, put those fields on the JSON object first
+  if (def.isEntry) {
+    cw.ln(`const inst = this._entryInfo.toJSON();`);
+  } else if (def.identifier.name !== 'EntryType') {
+    const url = `http://standardhealthrecord.org/spec/${def.identifier.namespace.replace('.', '/')}/${def.identifier.name}`;
+    cw.ln(`const inst = { 'shr.base.EntryType': { 'Value' : '${url}' } };`);
+  } else {
+    cw.ln(`const inst = {};`);
+  }
+
+  if (def.value !== undefined) {
+    cw.bl('if (this.value !== undefined)', () => {
+      if (def.value instanceof ChoiceValue) {
+        // Choices get a special treatment
+        cw.ln(`inst['Value'] = typeof this.value.toJSON === 'function' ? this.value.toJSON() : this.value;`);
```

Comment:
A `ChoiceValue` could also be a list, so you still need to handle that possibility here.  Unfortunately, it's a little trickier because it could be a list of primitives, a list of complex types, or a list containing a mix of primitives and complex types!